File: lib/core/rbac.dart
class RBAC {
  static bool canManageUsers(String role) {
    return role == "admin";
  }

  static bool canViewDashboard(String role) {
    return role == "admin";
  }

  static bool canManageStock(String role) {
    return role == "admin" || role == "pharmacy";
  }

  static bool canViewOrders(String role) {
    return role != "customer";
  }

  static bool isAdmin(String role) {
    return role == "admin";
  }
}

File: backend/core/rbac.py
ROLE_PERMISSIONS = {
    "admin": ["*"],
    "pharmacy_owner": ["manage_stock", "view_orders"],
    "supplier": ["view_orders"],
    "rider": ["update_delivery_status"],
    "customer": ["place_order"]
}


def has_permission(user, permission):
    perms = ROLE_PERMISSIONS.get(user.role, [])
    return "*" in perms or permission in perms


def is_admin(user):
    return user.role == "admin"
